multik-api/src/main/kotlin/org/jetbrains/kotlinx/multik/ndarray/operations/Add.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2020-2021 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package org.jetbrains.kotlinx.multik.ndarray.operations
+
+import org.jetbrains.kotlinx.multik.ndarray.data.*
+
+public inline fun <reified T : Number> D1Array<T>.append(n: D1Array<T>): D1Array<T>{
+    val totalSize :Int = this.size+n.size
+    var dest = arrayOfNulls<T>(totalSize)
+    System.arraycopy(Array<T>(this.size) { this.data[it] }, 0,dest, 0, this.size)
+    System.arraycopy(Array<T>(n.size){n.data[it]}, 0, dest,this.size, n.size)
```

Comment:
Better to use [primitive arrays](https://kotlinlang.org/docs/basic-types.html#primitive-type-arrays).
Use `dtype` and `when` to access primitive arrays, for example as done [here](https://github.com/Kotlin/multik/blob/cd26d4e7904ae9e9a574b64467ef7a93dc5d1beb/multik-jvm/src/main/kotlin/org/jetbrains/kotlinx/multik/jvm/JvmLinAlg.kt#L337)